back/app/controllers/concerns/locked_user_custom_fields_concern.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module LockedUserCustomFieldsConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def jsonapi_serializer_params_with_locked_fields
+    constraints = build_locked_custom_fields_constraints
+    jsonapi_serializer_params({ constraints: constraints })
+  end
+
+  def build_locked_custom_fields_constraints
+    return {} unless current_user
+
+    locked_custom_field_keys = verification_service.locked_custom_fields(current_user).map(&:to_s)
+    constraints = {}
+
+    # Find the corresponding codes for the locked custom field keys
+    # The serializer expects constraints keyed by 'code', not 'key'
+    CustomField.where(key: locked_custom_field_keys).each do |field|
+      code_key = field.code&.to_sym || field.key&.to_sym
```

Comment:
If the serializer only keys by code (which indeed it does) then locks are only allowed on built in fields and you should change this to:
```suggestion
      code_key = field.code&.to_sym
```